src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs

```diff
@@ -441,6 +440,14 @@ private async Task<List<string>> GetOrderedFilesToRestore()
             var fullBackupPath = GetBackupPath(backupPath);
             using (var zip = await GetZipArchiveForSnapshot(fullBackupPath))
             {
+                var voronDataDirectory = new VoronPathSetting(RestoreFromConfiguration.DataDirectory);
+                if (Directory.Exists(voronDataDirectory.FullPath) == false)
+                    Directory.CreateDirectory(voronDataDirectory.FullPath);
+
+                // validate free space
+                var snapshotSize = zip.Entries.Sum(entry => entry.Length);
```

Comment:
Does it download the whole snapshot file?

Reply:
At this point, you already downloaded it, yes. Because you have this opened.

Reply:
it downloads on [GetZipArchiveForSnapshot](https://github.com/ravendb/ravendb/pull/11023/files#diff-e08d344e1eb26f22609c114ab3008af6R441)